internal/http/services/owncloud/ocdav/token.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2018-2022 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package ocdav
+
+import (
+	"context"
+	"encoding/json"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"net/http"
+	"path"
+
+	gateway "github.com/cs3org/go-cs3apis/cs3/gateway/v1beta1"
+	user "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+	link "github.com/cs3org/go-cs3apis/cs3/sharing/link/v1beta1"
+	"github.com/cs3org/reva/v2/internal/http/services/owncloud/ocdav/spacelookup"
+	"github.com/cs3org/reva/v2/pkg/appctx"
+	ctxpkg "github.com/cs3org/reva/v2/pkg/ctx"
+	"github.com/cs3org/reva/v2/pkg/rgrpc/todo/pool"
+	"github.com/cs3org/reva/v2/pkg/rhttp/router"
+	"github.com/cs3org/reva/v2/pkg/utils"
+	"google.golang.org/grpc/metadata"
+)
+
+// TokenHandler handles requests for public link tokens
+type TokenHandler struct{}
+
+// Handler handles http requests
+func (t TokenHandler) Handler(s *svc) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		typ, tkn := router.ShiftPath(r.URL.Path)
+		tkn, _ = router.ShiftPath(tkn)
+
+		c, err := pool.GetGatewayServiceClient(s.c.GatewaySvc)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		switch typ {
+		case "protected":
+			s.handleGetToken(w, r, tkn, c, true)
+		case "unprotected":
+			s.handleGetToken(w, r, tkn, c, false)
+		}
+	})
+}
+
+// TokenInfo contains information about the token
+type TokenInfo struct {
+	// for all callers
+	Token             string `xml:"token"`
+	LinkURL           string `xml:"linkurl"`
+	PasswordProtected bool   `xml:"passwordprotected"`
+
+	// if not password protected
+	StorageID string `xml:"storageid"`
+	OpaqueID  string `xml:"opaqueid"`
+	Path      string `xml:"path"`
+
+	// if native access
+	SpacePath  string `xml:"spacePath"`
+	SpaceAlias string `xml:"spaceAlias"`
+	SpaceURL   string `xml:"spaceURL"`
+}
```

Comment:
hm, sometimes camelcase, sometimes all lowercase. Looking at the existing propfind, try sticking to all lowercase.

Reply:
to lowercase `TokenInfo` you can add
```
type TokenInfo struct {
	XMLName  xml.Name  `xml:"tokeninfo"`
```

Reply:
structs in ocs didn't have that file and are not xml specific, so I didn't add it yet. Recomment if I should.

Reply:
I used camelCase for json and alllowercase for xml. Does that sound correct?